internal/app/subsystems/api/http/http.go

```diff
@@ -48,16 +50,7 @@ func New(api api.API, config *Config) api.Subsystem {
 
 	// Authentication
 	authorized := r.Group("/")
-	if config.Auth.Username != "" || config.Auth.Password != "" {
-		util.Assert(config.Auth.Username != "", "http basic auth username is required")
-		util.Assert(config.Auth.Password != "", "http basic auth password is required")
-
-		accounts := gin.Accounts{
-			config.Auth.Username: config.Auth.Password,
-		}
-		basicAuthMiddleware := gin.BasicAuth(accounts)
-		authorized.Use(basicAuthMiddleware)
-	}
+	authorized.Use(BasicAuthMiddleware(config))
```

Comment:
I've noticed that specifying a nil value for `Use` leads to runtime issues so I think we need to conditionally set the middleware
```
if len(config.Auth) > 0 {
    authorized.Use(gin.BasicAuth(config.Auth))
}
```

Reply:
Indeed. I also removed the function ```BasicAuthMiddleware``` as i thought it wasnt needed at this juncture , after new changes, and , length would not be an issue as we would already our ```config``` initialised. Thus , in a scenario where no creds are provided , the ```config.Auth```  would still hold an initialised empty map .